_tasks/sync-proposal-data.mjs

```diff
@@ -0,0 +1,304 @@
+import { curlyQuote, loggerOpts } from './util/common.mjs';
+import { GhFileImporter } from './util/gh-file-importer.mjs';
+import { markdownTbl2json } from './util/table.mjs';
+import { mkdir, readFile, rmdir, writeFile } from 'fs/promises';
+import { ownProperty } from './util/object.mjs';
+import { ProposalRecord } from './types.mjs';
+import { resolve as pathResolve } from 'path';
+import he from 'he';
+import htmlStringify from 'rehype-stringify';
+import markdown from 'remark-parse';
+import raw from 'rehype-raw';
+import remark2rehype from 'remark-rehype';
+import sanitizeHtml from 'sanitize-html';
+import unified from 'unified';
+import visit from 'unist-util-visit';
+import yaml from 'js-yaml';
+
+const DIR_DATA = '_data';
+const DIR_TEMP = 'tmp';
+const REGEX_JS_INFO_STRS = /(javascript|js|mjs)/;
+const URL_RAW_PROPOSALS_README =
+  'https://raw.githubusercontent.com/tc39/proposals/HEAD/README.md';
+
+const require = createRequire(import.meta.url);
+
+const { prompt } = require('enquirer');
+const { Signale } = require('signale');
+const ghFileImporter = new GhFileImporter(DIR_TEMP);
+
+loggerOpts.scope = 'Sync Proposal Data Task';
+const log = new Signale(loggerOpts);
+
+await rmdir(DIR_TEMP, { recursive: true });
+await mkdir(DIR_TEMP);
+await ghFileImporter.importFileFromUrl(URL_RAW_PROPOSALS_README);
+
+// -----------------------------------------------------------------------------
+// Events
+// -----------------------------------------------------------------------------
+
+process.on('uncaughtException', (err) => {
+  log.error(err);
+});
+
+process.on('unhandledRejection', (wrn) => {
+  log.warn(process.stderr.fd, wrn);
+});
+
+// -----------------------------------------------------------------------------
+// Helpers
+// -----------------------------------------------------------------------------
+// > A 'full reference link' consists of a 'link text' immediately followed by a
+// > 'link label' that matches a link reference definition elsewhere in the
+// > document.
+// @see https://github.github.com/gfm/#full-reference-link
+//
+// [foo][bar]
+//   ^    ^----------- link label (value)
+//   |---------------- link text (key)
+//
+// @example ```markdown
+// [foo][bar]
+//
+
+// bar: http://foo.bar
+// ```
+
+/**
+ * Does it already have a description by reading the yaml from stage_3.yml
+ * @param {string}
+ * @returns {string}
+ */
+async function getDescription(prpslId) {
+  const data = await readFile(pathResolve(DIR_DATA, 'stage3.yml'), 'utf8');
+  let description;
+  yaml.safeLoadAll(data, (doc) => {
+    doc.forEach((value) => {
+      if (value.id === prpslId) {
+        // Proposal description has already has already been filled out.
+        log.info(
+          `Using pre-existing description for ${curlyQuote(prpslId)}: ${
+            value.description
+          }`
+        );
+        description = value.description;
+      }
+    });
+  });
+  if (!description) {
+    const response = await prompt({
```

Comment:
The only thing left to do on this PR is to figure out why this prompt doesn't display otherwise it's 100% done as far as I can tell. Would anyone happen to know why that is the case? You can test it by deleting the "description" field from one of the proposals in the YAML file and you will see that it hangs. Pressing ctrl+c unhangs it, but then the opportunity to enter a description is lost.

/cc @ljharb

Reply:
Hmm, it seems like a recent update to the proposals repo's README is messing up the `linkTextLike` algorithm. Marking as draft again until we find a better way to query for this info.

Reply:
Which update?

Reply:
Looks like the latest one (https://github.com/tc39/proposals/commit/d556f0acc866429c5ea5ac152abaa75128ed0452).

Reply:
Running the task now results in promise rejections and null data output to yaml file because it can't find the proposal repo's url the way it did before.

Reply:
The column widths changed, but that changes often - I’m not sure what else in that commit would have broken this.

Reply:
A few links were added to the bottom of the README. I know of a better way to go about this that should drastically simplify things.

Reply:
Ah, my mistake. I was able to track the problem down. Can disregard what I said above.